File: algorithm/杭电ACM/ambition0109/1286.java
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2011-04-11 21:19:47
////Compiler: Java
//////////////////////////////////////////////////////
////Problem ID: 1286
////Problem Title: 
////Run result: Accept
////Run time:1562MS
////Run memory:7136KB
//////////////////System Comment End//////////////////
import java.util.Scanner;

public class Main {
	static int GCD(int a,int b){
		while(b!=0){
			int t=b;
			b=a%b;
			a=t;
		}
		return a;
	}
	public static void main(String args[]){
		Scanner in=new Scanner(System.in);
		int t=in.nextInt();
		while(t--!=0){
			int n=in.nextInt();
			int cnt=0;
			for(int i=1;i<n;i++){
				if(GCD(n,i)==1) cnt++;
			}
			System.out.println(cnt);
		}
	}
}

File: algorithm/杭电ACM/ambition0109/1286.cpp
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2010-04-28 10:49:18
////Compiler: Visual C
//////////////////////////////////////////////////////
////Problem ID: 1286
////Problem Title: 
////Run result: Accept
////Run time:234MS
////Run memory:336KB
//////////////////System Comment End//////////////////
#include<stdio.h>
#include<math.h>
int num[32768];
main()
{
	int n,m,i,j,count;
	for(scanf("%d",&n);n>0;n--)
	{
		for(scanf("%d",&m),i=0;i<m;i++) num[i]=0;
		for(i=2;i<=m/2;i++)
			if(m%i==0)	num[i]=1;
		for(i=2;i<=m/2;i++)
		{
			if(num[i])
				for(j=2;i*j<m;j++)
					num[i*j]=1;
		}
		for(i=1,count=0;i<m;i++)
			if(!num[i]) count++;
		printf("%d\n",count);
	}
}